### ISSUE TITLE
Test runner reports all tasks as executed preventing correct UI examination

### ISSUE THREAD

**Issue by fcastill** [TIBCOSoftware/flogo-contrib#124]
Following the UI issue in TIBCOSoftware/flogo-web#421

When a flow is test-run the state service is reporting all the tasks in the flow instead of only the ones that were executed. Because of that the web UI cannot properly highlight the tasks that were executed.

I'm attaching a file with sequence of calls to the services and the flogo.json generated by the UI.
But the problem seems to be in the status reported to the state service, it returns something like the following json where entry with taskId "4" shouldn't be present as that task is linked to a branch with a condition that evals to false.

```
  {  
    "steps":[  
      {  
         "flow": {...},
          "id":"1",
          "taskId":"root",
          "tasks":null
      },
      {  
          "flow": { ... },
          "id":"2",
          "taskId":"2",
          "tasks":null
      },
      {  
          "flow":{ ... },
          "id":"3",
          "taskId":"5",
          "tasks":null
      },
      {  
          "flow":{ ... },
          "id":"4",
          "taskId":"4",
          "tasks":null
      },
      {  
          "flow":{ ... },
          "id":"5",
          "taskId":"3",
          "tasks":null
      }
    ]
  }
```

Calls sequence and flogo.json:

[flogo-steps-issue.zip](https://github.com/TIBCOSoftware/flogo-contrib/files/1334652/flogo-steps-issue.zip)

**Comment by mellistibco** [comment]
@fcastill this is no longer an issue, right?

**Comment by fcastill** [comment]
Not an issue anymore.

Fixed by TIBCOSoftware/flogo-services#14 and TIBCOSoftware/flogo-web#421
